impeller/entity/entity_pass_clip_stack.cc

```diff
@@ -145,6 +146,35 @@ EntityPassClipStack::ClipStateResult EntityPassClipStack::RecordClip(
     return result;
   }
 
+  // If the clip is an axis aligned rect and either is_aa is false or
+  // the clip is very nearly integral, then the depth write can be
+  // skipped for intersect clips. Since we use 4x MSAA, anything within
+  // about 0.2 of an integral value in either axis can be treated as
+  // approximately the same as an integral value.
+  if (!clip_coverage.is_difference_or_non_square &&
+      clip_coverage.coverage.has_value()) {
+    const Rect& coverage = clip_coverage.coverage.value();
+    constexpr Scalar threshold = 0.2;
+    if (!is_aa ||
+        (std::abs(std::round(coverage.GetLeft()) - coverage.GetLeft()) <=
+             threshold &&
+         std::abs(std::round(coverage.GetTop()) - coverage.GetTop()) <=
+             threshold &&
+         std::abs(std::round(coverage.GetRight()) - coverage.GetRight()) <=
+             threshold &&
+         std::abs(std::round(coverage.GetBottom()) - coverage.GetBottom()) <=
+             threshold)) {
+      subpass_state.clip_coverage.push_back(ClipCoverageLayer{
+          .coverage = Rect::Round(clip_coverage.coverage.value()),  //
+          .clip_height = previous_clip_height + 1                   //
```

Comment:
Don't we still need to track the clip coverage layer so that the scissor will get set during clip replays?

Reply:
Yeah good point, let me refactor this so I can just make `should_render` conditional.